editor/plugins/org.fusesource.ide.projecttemplates/src/org/fusesource/ide/projecttemplates/impl/simple/AbstractEAPSpringTemplate.java

```diff
@@ -49,11 +50,11 @@ public EAPSpringUnzipTemplateCreator(String suffix) {
 		private static final String TEMPLATE_SPRING = "template-medium-eap-wildfly-spring-fuse";
 		
 		@Override
-		public InputStream getTemplateStream(NewProjectMetaData metadata) throws IOException {
-			if(CamelDSLType.SPRING.equals(metadata.getDslType())) {
+		public InputStream getTemplateStream(CommonNewProjectMetaData metadata) throws IOException {
+			if (metadata instanceof ICamelDSLTypeSupport && CamelDSLType.SPRING.equals((((ICamelDSLTypeSupport)metadata).getDslType()))) {
 				return getTemplateStream(String.format("%s%s%s.zip", TEMPLATE_FOLDER, TEMPLATE_SPRING, suffix));
 			}
-			return null;
+			throw new IOException("Invalid project metadata not supporting Camel DSL types");
```

Comment:
Method is already defined as throws IOException, so callers handle errors of that type accordingly. Also if the metadata is not valid then we can't read/write the template which is also a kind of IOException even if no direct IO is involved. What is your suggestion here? To change the signature of that method and have all other templates affected too?

Reply:
I think I would let the return null. it will throws the NPE but in theory only the developers would encounter it when developing a new Template.
The cleaner solution would be effectively to launch a specific check exception even if it is affecting all AbstractTemplates. (and by the way, I think that it is because we have some code quite duplicated that it will affect several places, otherwise we should be able to modify at one or 2 places only, I reported an issue for that few hours ago :-) https://issues.jboss.org/browse/FUSETOOLS-2745 )
Another solution would be to launch a RuntimeException.

So we have 4 alternatives. for this PR I vote in this order:
1. continue to return null
2. Throw RuntimeException (it adds a little value with an explanation why it fails and don't lead to think that there is an issue with I/O)
3. Throw IOException as you did as it is a checked exception which adds an explanation and avoid changing signature
4. throw a dedicated checked exception as it requires too much work for the added value compared to the other solution.

What do you think? Do you see other benefits drawbacks?

Reply:
tbh I find it safer to throw an exception defined in the method via "throws" instead of returning *null* which can have unexpected impacts as the called might not have handled it. The caller IS FORCED to handle the thrown exception.

Reply:
I'm not sure that using the IOException is not adding more confusion than it is helping.
I agree that it is safer to throw a checked exception but it needs to be a dedicated exception to provide a clear improved value than returning null

Reply:
I think from a users point of view the kind of the exception is less interesting. What users look for is the message that the exception shows. Anyway from a technical pov you are right. I will add a dedicated exception for this error.